doc/source/ray-core/examples/map_reduce.ipynb

```diff
@@ -0,0 +1,426 @@
+  },
+  {
+   "cell_type": "markdown",
+   "id": "52009879",
+   "metadata": {},
+   "source": [
+    "We will use the apply_map function on a large collection of documents by marking it as a task in Ray using the `@ray.remote` decorator.\n",
+    "When we call `apply_map`, it will be applied to three sets of document data (`num_partitions=3`).\n",
+    "The `apply_map` function will return three lists, one for each partition.\n",
+    "We do this so that Ray can rearrange the results of the map phase and distribute them to the appropriate nodes for us."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "id": "a2fed469",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import ray\n",
+    "\n",
+    "@ray.remote\n",
+    "def apply_map(corpus, num_partitions=3):\n",
+    "    map_results = [list() for _ in range(num_partitions)]\n",
+    "    for document in corpus:\n",
+    "        for result in map_function(document):\n",
+    "            first_letter = result[0].decode(\"utf-8\")[0]\n",
+    "            word_index = ord(first_letter) % num_partitions\n",
+    "            map_results[word_index].append(result)\n",
+    "    return map_results"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a1ba13f8",
+   "metadata": {},
+   "source": [
+    "For text corpora that can be stored on a single machine, it is unnecessary to use the map phase.\n",
+    "However, when the data needs to be divided across multiple nodes, the map phase becomes useful.\n",
+    "In order to apply the map phase to our corpus in parallel, we use a remote call on `apply_map`, just like we have done in previous examples.\n",
+    "The main difference now is that we also specify that we want three results returned (one for each partition) using the num_returns argument."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "id": "360b19b8",
+   "metadata": {
+    "lines_to_next_cell": 2
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Mapper 0, return value 0: [(b'of', 1), (b'is', 1)]\n",
+      "Mapper 0, return value 1: [(b'python,', 1), (b'peters', 1)]\n",
+      "Mapper 0, return value 2: [(b'the', 1), (b'zen', 1)]\n",
+      "Mapper 1, return value 0: [(b'unless', 1), (b'in', 1)]\n",
+      "Mapper 1, return value 1: [(b'although', 1), (b'practicality', 1)]\n",
+      "Mapper 1, return value 2: [(b'beats', 1), (b'errors', 1)]\n",
+      "Mapper 2, return value 0: [(b'is', 1), (b'is', 1)]\n",
+      "Mapper 2, return value 1: [(b'although', 1), (b'a', 1)]\n",
+      "Mapper 2, return value 2: [(b'better', 1), (b'than', 1)]\n"
+     ]
+    }
+   ],
+   "source": [
+    "map_results = [\n",
+    "    apply_map.options(num_returns=num_partitions)\n",
+    "    .remote(data, num_partitions)\n",
+    "    for data in partitions\n",
+    "]\n",
+    "\n",
+    "for i in range(num_partitions):\n",
+    "    mapper_results = ray.get(map_results[i])\n",
```

Comment:
I believe currently this will pull all the data onto the driver, right? I think that's fine for the simple example, but we should consider adding that as a caveat and giving a hint as to how that could be fixed :)

Reply:
right, very good point!

Reply:
Added this line: ```For simplicity, this example collects all the data onto the driver. The examples in the following section demonstrate how to divide data across nodes.```
@maxpumperla, please make any necessary corrections.